Add column for collection to tools, organisation and contributors.

koechkevin opened the pull request:
## Description

There's a column we missed during our implementation called collection, this PR adds that column to Ecosystem Settings, Airtable process Data and an ability to filter by the same column via block settings.

Since we cannot name a column as `collection` in payload, I have used `{entity}Collection` naming standard.


## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)


## Screenshots
![image](https://github.com/CodeForAfrica/ui/assets/39160236/ce170334-e9ff-46fe-9fce-0964057f9df5)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

kilemensi commented:
Ola @kelvinkipruto 

1. Where are we showing the fitler?
2. It looks broken to me (i.e. appears on a new line below other filters). Shouldn't all align on one row (Do we have updated designs that include this filter?

kilemensi commented:
> Ola @koechkevin
> 
>     1. Where are we showing the filter?
> 
>     2. It looks broken to me (i.e. appears on a new line below other filters). Shouldn't all align on one row (Do we have updated designs that include this filter?

koechkevin commented:
> Ola @koechkevin
> 
> 1. Where are we showing the filter?
> 2. It looks broken to me (i.e. appears on a new line below other filters). Shouldn't all align on one row (Do we have updated designs that include this filter?

@kilemensi  I added this to blocks to select if it is required to appear as a filter.

<img width="950" alt="image" src="https://github.com/CodeForAfrica/ui/assets/39160236/09756b0d-a974-4a08-b474-134fa4a0b5df">

kilemensi commented:
> > Ola @koechkevin
> > 
> > 1. Where are we showing the filter?
> > 2. It looks broken to me (i.e. appears on a new line below other filters). Shouldn't all align on one row (Do we have updated designs that include this filter?
> 
> @kilemensi I added this to blocks to select if it is required to appear as a filter.
> 

Cool @koechkevin, my question was how do we prevent things like

![image](https://github.com/CodeForAfrica/ui/assets/39160236/ce170334-e9ff-46fe-9fce-0964057f9df5)

i.e. selection spilling over multiple lines?

kilemensi commented on apps/charterafrica/src/lib/data/common/processPageContributors.js:
```diff
@@ -8,7 +8,7 @@ import formatDateTime from "@/charterafrica/utils/formatDate";
 
 const queryBuilder = (query) => {
-  const { search, location } = query;
+  const { search, location, contributorCollection } = query;
```
Why no just `collection` (similar to just `location`)?

kilemensi commented on apps/charterafrica/src/components/EcosystemList/EcosystemFilter.js:
```diff
@@ -8,7 +8,7 @@ import { ControlledSearchInput } from "@/charterafrica/components/SearchInput";
 const EcosystemFilter = React.forwardRef(function EcosystemFilter(props, ref) {
   const { onChange, searchPlaceholder, values, filterOptions = [] } = props;
-
+  const spans = Math.floor(9 / filterOptions.length);
```
Add a comment to explain why `9`

kilemensi commented on apps/charterafrica/src/lib/data/common/processPageContributors.js:
```diff
@@ -8,7 +8,7 @@ import formatDateTime from "@/charterafrica/utils/formatDate";
 
 const queryBuilder = (query) => {
-  const { search, location } = query;
+  const { search, location, classification } = query;
```
The data team confirmed collection === classification?

koechkevin replied:
@kilemensi 

From this [document](https://docs.google.com/document/d/1HDQBaOJD8XjtJfo-RR5s-TOdh1njC88ZmDqUVNeJwxg/edit?usp=sharing),
<img width="779" alt="image" src="https://github.com/CodeForAfrica/ui/assets/39160236/49377c87-9d59-49b7-bbf3-4de211f7e8f5">

kilemensi replied:
Thanks @koechkevin ... If I understand the doc, `classification` is the _source_ for `collection`, it doesn't mean `classification` **is** `collection`. Similar to `DB` is the source for `Tool` but you can't say `DB` **is** `Tool`.

But anyways, let's get it over the line and push on academy.AFRICA.
